Kernel/Arch/riscv64/PCI/Initializer.cpp

```diff
@@ -180,24 +182,61 @@ void initialize()
                     auto prefetchable = (pci_address_metadata >> OpenFirmwareAddress::prefetchable_offset) & OpenFirmwareAddress::prefetchable_mask;
                     if (prefetchable)
                         continue; // We currently only use non-prefetchable 32-bit regions, since 64-bit regions are always prefetchable - TODO: Use 32-bit prefetchable regions if only they are available
-                    if (pci_32bit_mmio_base.has_value() && pci_32bit_mmio_size >= mmio_size)
+                    if (pci_32bit_mmio_size >= mmio_size)
                         continue; // We currently only use the single largest region - TODO: Use all available regions if needed
                     pci_32bit_mmio_base = mmio_address;
                     pci_32bit_mmio_size = mmio_size;
                 } else {
-                    if (pci_64bit_mmio_base.has_value() && pci_64bit_mmio_size >= mmio_size)
+                    if (pci_64bit_mmio_size >= mmio_size)
                         continue; // We currently only use the single largest region - TODO: Use all available regions if needed
                     pci_64bit_mmio_base = mmio_address;
                     pci_64bit_mmio_size = mmio_size;
                 }
             }
         }
+
+        auto maybe_interrupt_map = node.get_property("interrupt-map"sv);
+        auto maybe_interrupt_map_mask = node.get_property("interrupt-map-mask"sv);
+        if (maybe_interrupt_map.has_value() && maybe_interrupt_map_mask.has_value()) {
+            auto mask_stream = maybe_interrupt_map_mask.value().as_stream();
+            u32 metadata_mask = MUST(mask_stream.read_value<BigEndian<u32>>());
+            MUST(mask_stream.discard(sizeof(u32) * 2));
```

Comment:
Could you please add a `VERIFY` that #interrupt-cells of the pci node is 1?

Reply:
Why? that's discarding the address part of the PCI address, I don't see why the interrupt-cells would affect it?

Reply:
Oh sorry, I meant to comment on the next line, where your extract the pin mask.

Reply:
The PCI specification defines it as an 8-bit field, how could it take more than 1 word?

Reply:
I just wanted to be extra sure that the devicetree contains the data we expect.
We also verify *-cells properties in other places :shrug:

Reply:
I guess it doesn't hurt, I'll add it.